packages/katello/katello/hostname-change.rb

```diff
@@ -345,6 +345,53 @@ def run
       STDOUT.puts "updating hostname in foreman installer scenarios"
       self.run_cmd("sed -i -e 's/#{@old_hostname}/#{@new_hostname}/g' #{scenarios_path}/*.yaml")
 
+      if @scenario_answers['foreman_proxy']['dns']
+        STDOUT.puts 'gathering DNS data...'
+        local_ip = '127.0.0.1'
+        zone = @scenario_answers['foreman_proxy']['dns_zone']
+        old_fqdn = @old_hostname
+        soa_admin_domain = "root.#{zone}"
+        zone_file = "/var/named/dynamic/db.#{zone}"
+        zone_file_data = File.read(zone_file).split("\n")
+        a_record = zone_file_data.find { |line| line.include?(' A ') }
+        existing_ip = a_record.split(' A ').last if a_record
+        serial_line = zone_file_data.find { |line| line.include?('Serial') }
+        existing_serial = serial_line.split('Serial').first.to_i if serial_line
+        new_serial = (existing_serial + 1).to_s if existing_serial
+        new_fqdn = @new_hostname
+        key_file = '/etc/rndc.key'
+        reverse_zone = @scenario_answers['foreman_proxy']['dns_reverse'].first
+
+        dns_data = {
+          local_ip: local_ip,
```

Comment:
There certainly is no harm in assigning and referencing here, but seems like doing `local_ip: '127.0.0.1'` will be fine if its not referenced anywhere else. Same goes for some other values in the hash. Building this hash feels like a good candidate for its own method

Reply:
@johnpmitsch My idea was that I'll gather every value I'm about to give to nsupdate, and make sure all of them are present.  `local_ip` will always be present, so I guess I included it just for completeness.  It's not strictly necessary here I suppose.